src/main/java/seedu/address/logic/parser/Parser.java

```diff
@@ -84,15 +80,19 @@ public Command parseCommand(String userInput) {
      * @return the prepared command
      */
     private Command prepareAdd(String args){
-        final Matcher matcher = FLOATING_TASK_ARGS_FORMAT.matcher(args.trim());
+        final Matcher matcher = KEYWORDS_ARGS_FORMAT.matcher(args.trim());
         // Validate arg string format
         if (!matcher.matches()) {
             return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
         }
+        final String keywords = matcher.group("keywords").trim();
         try {
-            return new AddCommand(
-                    matcher.group("name"),
-                    matcher.group("priorityValue"));
+            if (keywords.contains("rank")) {
+                return new AddCommand(keywords.substring(0, keywords.indexOf("rank")),
+                        keywords.substring(keywords.indexOf("rank")+4, keywords.length()));
+            }
+            else
+                return new AddCommand(keywords, null);
```

Comment:
can u create a new add command constructor that uses the floating task constructor with only 1 var